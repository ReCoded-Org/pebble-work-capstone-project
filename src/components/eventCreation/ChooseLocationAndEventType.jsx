import React, { useState } from "react";

import Button from "../Button";
import InputComponent from "../InputComponent";

const ChooseLocationAndEventType = () => {
    const [Location, setLocation] = useState("Which city?");
    return (
        <>
            <div className='mx-8'>
                <div className='flex flex-col   gap-10 py-3 md:flex-row md:justify-between'>
                    <div>
                        <h1 className='text-3xl font-bold text-secondary-300'>
                            First, set your eventsâ€™s location:
                        </h1>
                        <p className='my-1 w-full text-gray-500'>
                            Pebble events can be both local or online. Choose
                            where you want to host your event.
                        </p>
                        <div className='relative w-full py-5'>
                            <div className='pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3'>
                                <svg
                                    aria-hidden='true'
                                    className='h-5 w-5 text-gray-500 dark:text-gray-400'
                                    fill='currentColor'
                                    viewBox='0 0 20 20'
                                    xmlns='http://www.w3.org/2000/svg'
                                >
                                    <path
                                        fillRule='evenodd'
                                        d='M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z'
                                        clipRule='evenodd'
                                    ></path>
                                </svg>
                            </div>
                            <InputComponent
                                placeholder='Search for a city'
                                className='w-full rounded border border-primary-200 px-9 py-1 shadow focus:outline-none'
                                callback={(event) => {
                                    {
                                        event.target.value.length === 0
                                            ? setLocation("Which city?")
                                            : setLocation(event.target.value);
                                    }
                                }}
                            />
                        </div>
                    </div>
                    <div className='flex w-full items-center justify-center rounded-full bg-sky-100'>
                        <div className='my-10 flex items-center justify-center  text-6xl font-bold text-primary-200 '>
                            {Location}
                        </div>
                    </div>
                </div>
                <div className='flex flex-col  gap-4  py-3 '>
                    <div>
                        <h1 className='text-3xl font-bold text-secondary-300'>
                            Second, Choose event type:
                        </h1>
                        <p className='my-1  w-full text-gray-500 '>
                            Every pebble event should serve at least one of the
                            sustainable development goals of United Nations.
                            Which goal do you want to help in? Select all that
                            apply.
                        </p>
                    </div>
                    <div className='md:flex md:justify-between'>
                        <div className='md:flex md:flex-col'>
                            <Button
                                label='No Poverty'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52'
                                hoverBgColor='bg-primary-200 border hover:text-white '
                            />
                            <Button
                                label='Zero Hunger'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52'
                                hoverBgColor='bg-primary-200 border hover:text-white'
                            />
                            <Button
                                label='Clean Water and Sanitation'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52'
                                hoverBgColor='bg-primary-200 border hover:text-white'
                            />
                            <Button
                                label='Gender Equality'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52'
                                hoverBgColor='bg-primary-200 border hover:text-white'
                            />
                        </div>
                        <div className='md:flex md:flex-col'>
                            <Button
                                label='Industry, Innovation, and Infrastructure'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52'
                                hoverBgColor='bg-primary-200 border hover:text-white'
                            />

                            <Button
                                label='Reduced Inequalities'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white'
                            />
                            <Button
                                label='Climate Action'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white'
                            />
                            <Button
                                label='Life Below Water'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white'
                            />
                        </div>
                        <div className='md:flex md:flex-col'>
                            <Button
                                label='Life on Land'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white '
                            />
                            <Button
                                label='Quality Educationg'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white hover:border-primary-200 hover:border-primary-200'
                            />
                            <Button
                                label='Sustainable Cities and Communities'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white '
                            />
                            <Button
                                label='Good Health and 
                            Well-being  '
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white '
                            />
                        </div>{" "}
                        <div className='md:flex md:flex-col'>
                            <Button
                                label='Decent Work and Economic Growth'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white '
                            />{" "}
                            <Button
                                label='Responsible Consumption/Production'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white '
                            />
                            <Button
                                label='Peace, Justice and Strong Institutions'
                                borderColor='border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white '
                            />
                            {/* <Button
                                label='Affordable and 
                            Clean Energy'
                                borderColor='border-primary-200 w-full rounded-full md:w-52
                                '
                                hoverBgColor='bg-primary-200 border hover:text-white '
                            /> */}
                        </div>
                    </div>
                </div>
            </div>
        </>
    );
};

export default ChooseLocationAndEventType;
