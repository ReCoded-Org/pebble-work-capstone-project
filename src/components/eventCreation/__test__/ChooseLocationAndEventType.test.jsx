import renderer from "react-test-renderer";

import ChooseLocationAndEventType from "../ChooseLocationAndEventType";

it("renders correctly", () => {
    const tree = renderer.create(<ChooseLocationAndEventType />).toJSON();
    expect(tree).toMatchSnapshot();
});
