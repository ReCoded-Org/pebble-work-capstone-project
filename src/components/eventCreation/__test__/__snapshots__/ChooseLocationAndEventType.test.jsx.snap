// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mx-8"
>
  <div
    className="flex flex-col gap-10 py-3 md:flex-row md:justify-between"
  >
    <div>
      <h1
        className="text-3xl font-bold text-secondary-300"
      >
        First, set your eventsâ€™s location:
      </h1>
      <p
        className="my-1 w-full text-gray-500"
      >
        Pebble events can be both local or online. Choose where you want to host your event.
      </p>
      <div
        className="relative w-full py-5"
      >
        <div
          className="pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3"
        >
          <svg
            aria-hidden="true"
            className="h-5 w-5 text-gray-500 dark:text-gray-400"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <input
          className="w-full rounded border border-primary-200 px-9 py-1 shadow focus:outline-none"
          onChange={[Function]}
          placeholder="Search for a city"
          type="text"
        />
        <label />
      </div>
    </div>
    <div
      className="flex w-full items-center justify-center rounded-full bg-sky-100"
    >
      <div
        className="my-10 flex items-center justify-center text-6xl font-bold text-primary-200 "
      >
        Which city?
      </div>
    </div>
  </div>
  <div
    className="flex flex-col gap-4 py-3 "
  >
    <div>
      <h1
        className="text-3xl font-bold text-secondary-300"
      >
        Second, Choose event type:
      </h1>
      <p
        className="my-1 w-full text-gray-500 "
      >
        Every pebble event should serve at least one of the sustainable development goals of United Nations. Which goal do you want to help in? Select all that apply.
      </p>
    </div>
    <div
      className="md:flex md:justify-between"
    >
      <div
        className="md:flex md:flex-col"
      >
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white  undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52 hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          No Poverty
        </button>
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52 hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Zero Hunger
        </button>
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52 hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Clean Water and Sanitation
        </button>
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52 hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Gender Equality
        </button>
      </div>
      <div
        className="md:flex md:flex-col"
      >
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52 hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Industry, Innovation, and Infrastructure
        </button>
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52  hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Reduced Inequalities
        </button>
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52  hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Climate Action
        </button>
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52  hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Life Below Water
        </button>
      </div>
      <div
        className="md:flex md:flex-col"
      >
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white  undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52  hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Life on Land
        </button>
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white hover:border-primary-200 hover:border-primary-200 undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52  hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Quality Educationg
        </button>
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white  undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52  hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Sustainable Cities and Communities
        </button>
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white  undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52  hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Good Health and Well-being 
        </button>
      </div>
       
      <div
        className="md:flex md:flex-col"
      >
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white  undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52  hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Decent Work and Economic Growth
        </button>
         
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white  undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52  hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Responsible Consumption/Production
        </button>
        <button
          className="m-2 undefined undefined undefined  hover:bg-primary-200 border hover:text-white  undefined undefined  py-1  px-2 border-secondary-300 w-full rounded-full hover:border-primary-200 md:w-52  hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          Peace, Justice and Strong Institutions
        </button>
      </div>
    </div>
  </div>
</div>
`;
